catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -305,6 +313,43 @@ private static void loadLegoEV3SettingsFromProject(Project project, Context cont
 		return conflicts;
 	}
 
+	public static synchronized boolean checkIfURLIsInWhitelist(String url) {
+		if (urlWhitelistPattern == null) {
+			try {
+				initializeURLWhitelistPattern();
+			} catch (IOException | JSONException | NullPointerException e) {
+				Log.e(TAG, "Cannot read URL whitelist.", e);
+				return false;
+			}
+		}
+		return urlWhitelistPattern.matcher(url).matches();
+	}
+
+	private static void initializeURLWhitelistPattern() throws IOException, JSONException, NullPointerException {
+		InputStream stream = Utils.getInputStreamFromAsset(instance.applicationContext, Constants.URL_WHITELIST_JSON_FILE_NAME);
+		JSONObject whiteList = Utils.getJsonObjectFromInputStream(stream);
+		JSONArray domains = whiteList.getJSONArray(Constants.URL_WHITELIST_JSON_ARRAY_NAME);
+
+		StringBuilder trustedDomains = new StringBuilder("(");
+		for (int i = 0; i < domains.length(); i++) {
+			trustedDomains.append(domains.getString(i));
+
+			if (i < domains.length() - 1) {
+				trustedDomains.append('|');
+			}
+		}
+		trustedDomains.append(')');
+
+		urlWhitelistPattern = Pattern.compile("https?://([a-zA-Z-.:@]+\\.)?"
+				+ trustedDomains.toString().replaceAll("\\.", "\\\\.")
+				+ "(:[0-9]{1,5})?(/[a-zA-Z0-9-()@:%_\\\\+~#.?&/=]*)?");
```

Comment:
```suggestion
		urlWhitelistPattern = Pattern.compile("https?://([a-zA-Z0-9-]+\\.)*"
				+ trustedDomains.toString().replaceAll("\\.", "\\\\.")
				+ "(:[0-9]{1,5})?(/.*)?");
```
After finding some examples that did not work because numbers incorrectly were not allowed at the front (e.g, connect4.ist.tugraz.at) and understanding that https://www.ietf.org/rfc/rfc1738.txt has been superceded by the newer https://tools.ietf.org/html/rfc3986 (also see https://en.wikipedia.org/wiki/Basic_access_authentication ) meaning that passwords have been depreciated in https? requests, and that basically any character can follow in the path/query/fragment part of the URI after the host and option port number.